src/amo/reducers/home.js

```diff
@@ -8,49 +8,113 @@ import {
 import { SET_CLIENT_APP } from 'core/constants';
 import { createInternalAddon } from 'core/reducers/addons';
 import { isTheme } from 'core/utils';
-import type { AddonType, ExternalAddonType } from 'core/types/addons';
 import type { SetClientAppAction } from 'core/actions';
+import type {
+  AddonType,
+  ExternalAddonType,
+  PartialExternalAddonType,
+} from 'core/types/addons';
+
+export const FETCH_HOME_DATA: 'FETCH_HOME_DATA' = 'FETCH_HOME_DATA';
+export const LOAD_HOME_DATA: 'LOAD_HOME_DATA' = 'LOAD_HOME_DATA';
+
+export type PrimaryHeroShelfExternalType = {|
+  id: string,
+  guid: string,
+  homepage: string,
+  name: string,
+  type: string,
+|};
+
+export type ExternalPrimaryHeroShelfType = {|
+  gradient: {|
+    start: string,
+    end: string,
+  |},
+  featured_image: string,
+  description: string | null,
+  addon: PartialExternalAddonType | void,
```

Comment:
It's tricky because either `addon` or `external` will be present. I would use `optional` here instead of `void`:

```
   ...
   addon?: PartialExternalAddonType,
   external?: PrimaryHeroShelfExternalType
```